### PR TITLE
Support a reference filter types

### PR DESCRIPTION
**Pull request by saimonation**
- Support listing device logs from Portal
- Support adding custom filters to logs module
- Update UT

### CODE REVIEW

File: cterasdk/core/query.py [hunk 1]
```diff
@@ -104,7 +112,7 @@ def after(self, value):
         return self.setValue(value)
 
     def setValue(self, value):
-        self.filter._classname = FilterType.fromValue(value)  # pylint: disable=protected-access,attribute-defined-outside-init
+        self.filter._classname = FilterType.fromValue(value, self.reference)  # pylint: disable=protected-access,W0201
```
Comment (ygalblum) [inline cterasdk/core/query.py:115]: Why do we need the change in the pylint disable list?

File: cterasdk/core/logs.py [hunk 2]
```diff
@@ -11,28 +11,70 @@ class Logs(BaseCommand):
...
+        :param str,optional before: Get logs before this date (in format "%m/%d/%Y %H:%M:%S"), defaults to None
+        :param str,optional after: Get logs after this date (in format "%m/%d/%Y %H:%M:%S"), defaults to None
+        :param list[cterasdk.core.query.FilterBuilder],optional filters: List of additional filters, defaults to None
+
+        :return: Iterator for all matching logs
+        :rtype: cterasdk.lib.iterator.Iterator[cterasdk.object.Object]
+        """
+        return self.logs(topic, min_severity, enum.OriginType.Device, name, before, after, filters)
+
+    def logs(self,
```
Comment (ygalblum) [inline cterasdk/core/logs.py:37]: Why do we need the name change here?

Comment (saimonation) [reply cterasdk/core/logs.py:37]: ah i don't like the inconsistency we had with camel case vs the pythonic way of naming fields. Old habits from coding java

File: cterasdk/core/query.py (hunk 1 above)
Comment (saimonation) [reply cterasdk/core/query.py:115]: The line was longer than 150 chars :(

File: cterasdk/core/logs.py (hunk 2 above)
Comment (ygalblum) [reply cterasdk/core/logs.py:37]: Oh, I meant the method name get->logs (I got the parameter names)

Comment (saimonation) [reply cterasdk/core/logs.py]: Not for any good reason. I just thought it's more aligned with core/devices. But I just changed it back. See update

Comment (saimonation) [reply cterasdk/core/logs.py]: @ygalblum - I changed it back, lmk if you approve

**Review by ygalblum — APPROVED** [review]

**Pull request merged**